packages/git/change.js

```diff
@@ -89,23 +89,30 @@ class Change {
       mergeCommit = await this._makeMergeCommit(newCommit, commitOpts);
 
       try {
-        if(this.fetchOpts) {
+        if (this.fetchOpts) {
           await this._pushCommit(mergeCommit);
         } else {
           await this._applyCommit(mergeCommit);
         }
       } catch (err) {
-        await delay(delayTime);
         log.warn('Failed to finalize commit "%s"', err);
 
+        if (this.fetchOpts) {
+          // pull remote before allowing process to continue
+          await this.repo.fetchAll(this.fetchOpts);
```

Comment:
This is a network request that is likely to throw exceptions sometimes. If it does, we will break out of the retry loop entirely.

Perhaps we can move the fetchAll so it happens before `_makeMergeCommit`, but only when a `needsFetchAll` flag is set. The flag would start out as false but set to true inside the `catch` block.

Reply:
Yep that sounds like a good idea 👍 I'll do that now